### ISSUE TITLE
[RN Android] JVM mismatch in react native projects using latest AGP

### ISSUE THREAD

**Issue by opxyd** [Tealium/tealium-react-native#170]
On trying to add the Tealium SDK (following the doc available at https://docs.tealium.com/platforms/react-native/install/), in the a new React Native project (using react native version 0.73.3). the below error occurs:

```
Execution failed for task ':tealium-react-native:compileDebugKotlin'.
> 'compileDebugJavaWithJavac' task (current target is 17) and 'compileDebugKotlin' task (current target is 11)
jvm target compatibility should be set to the same Java version.
```

It's not possible (or ideal) to run the project using Java 11 since the gradle plugin require Java 17.

Steps to reproduce:
1. Create a fresh new react native project following the official doc.
2. Install tealium-react-native from the official docs
3. Run the project

**Comment by opxyc** [comment]
If there are any suggested solutions, please do share, it will be really helpful. Thanks in advance.

**Comment by craigrouse** [comment]
@opxyc We've been testing a fix for this to make sure it doesn't impact earlier RN versions. We hope to finish testing early next week, then we'll make a new release.

**Comment by opxyc** [comment]
Thanks a lot for looking into this @craigrouse. Really appreciate your effort.

**Comment by opxyc** [comment]
Thanks a lot for looking into this @craigrouse. Really appreciate your effort.

**Comment by jameskeith** [comment]
This should now be resolved in the [latest release 2.4.0](https://github.com/Tealium/tealium-react-native/releases/tag/2.4.0)
